openvm/src/lib.rs

```diff
@@ -587,49 +593,46 @@ pub fn get_pc_idx_count(guest: &str, guest_opts: GuestOptions, inputs: StdIn) ->
 }
 
 pub fn instructions_to_airs<P: IntoOpenVm, VC: VmConfig<OpenVmField<P>>>(
-    exe: VmExe<OpenVmField<P>>,
     vm_config: VC,
+    used_instructions: impl Iterator<Item = VmOpcode>,
 ) -> BTreeMap<usize, SymbolicMachine<P>>
 where
     VC::Executor: Chip<BabyBearSC>,
     VC::Periphery: Chip<BabyBearSC>,
 {
-    let mut chip_complex: VmChipComplex<_, _, _> = vm_config.create_chip_complex().unwrap();
-    exe.program
-        .instructions_and_debug_infos
-        .iter()
-        .map(|instr| instr.as_ref().unwrap().0.opcode)
-        .unique()
+    let chip_complex: VmChipComplex<_, _, _> = vm_config.create_chip_complex().unwrap();
+    
+    // Note that we could use chip_complex.inventory.available_opcodes() instead of used_instructions,
```

Comment:
This function is only available if we merge https://github.com/powdr-labs/openvm/pull/22

I think we should either merge it or remove this comment.

I think it does make sense to have it, and it seems outrageous that keeping the Keccak precompile in memory causes memory issues. I think we should eventually fix that and then include all the available opcodes.

Reply:
sure let's merge the other PR then